### ISSUE TITLE
Cannot upload images anymore

### ISSUE THREAD

**Issue by ME-Merlin2010** [scottrice/ConsoleGrid#36]
Hi, 
I have an account on ConsoleGrid and a few weeks ago I could upload some images. Last few day everytime I try to upload some images the following message appear: 
"There was an error uploading your image. Please try again, or try a different image"
Could you check this behavior?
Greetings Tobias

**Comment by RuiOrey** [comment]
same here

**Comment by demonblah** [comment]
Ditto! can't upload anything. Thought maybe I was doing something wrong so glad to see others are having the same issue. Is this site still being working on by anyone?

**Comment by TechnoFlawgic** [comment]
Just wanna add my name to the list, also I'm pretty certain some images have fallen off completely. See: The Legend of Zelda - Oracle of Ages

**Comment by dkcas11** [comment]
Same here... I've added a couple of games, which just need images. Just for the convenience for ICE to locate the games and actually create them in Steam. I have the images on my desktop though, ready to go :)

**Comment by imkrut** [comment]
Same here

**Comment by reploid** [comment]
I still can't upload either.

**Comment by Schroedingers-Cat** [comment]
Are there any alternatives we can use in conjunction with ice?

**Comment by giacomocerquone** [comment]
I signed up right now and at the moment I can't upload an image for Super Mario 64.

UPDATE
Anyway there shouldn't be problem with the code considering that his last commit is dated 4 Jan. It could be some configuration problem, unluckily I don't know ruby and I can't fix it.
He is busy for sure, but I'd like to get his attention (hoping he doesn't mind) @scottrice

**Comment by MJLHThomassen** [comment]
Same for me, can't upload images. I also accidently marked a game (Chrono Cross - CD1 for Playstation) as duplicate and it automatcially removed it. Now i re-added it but i cant add the image back :(

**Comment by nateberkopec** [comment]
Here's an update.

@scottrice has let me help out with ConsoleGrid. This is definitely the first thing I want to fix. However, like @giacomocerquone said, this is probably a config issue, not a code issue. I'm still waiting on access to the server so I can take a look.

**Comment by dkcas11** [comment]
Awesome Nate, that is great news :)

**Comment by asc6** [comment]
@nateberkopec any update? Still gives an error on uploading images.

**Comment by nateberkopec** [comment]
Waiting on @scottrice for credentials

**Comment by asc6** [comment]
Gotcha! Thanks!

**Comment by ferdbold** [comment]
Can't upload images either.

**Comment by Schroedingers-Cat** [comment]
@nateberkopec did you get the credentials for the server?

**Comment by AetherCollective** [comment]
Same here. Made an account but can't upload images for any games, including Megaman Star Force and Wind Waker

**Comment by FriendsNone** [comment]
As of today (12/23/2016), still can't upload any images. I have a image for Zelda: Four Swords Adventure. Hope this problem will be fixed :)

**Comment by coryallegory** [comment]
It's been a year since the last repo update, I wonder if the website has run out of space for image uploads in that time....

**Comment by Schroedingers-Cat** [comment]
@scottrice @nateberkopec you wrote you were onto this issue monts ago. What's the status?

**Comment by robertpallister** [comment]
@scottrice  @nateberkopec  any news on this? Trying to setup ice, I got everything working but the images.

**Comment by billkie** [comment]
Any update on this?

**Comment by ScottGotRekt** [comment]
To echo the previous comment, any updates?

Attempting to upload  picture for a title called Vampire Night for the PS2, previously hadn't been added.

I need muh thumbnails.

**Comment by hodgeac** [comment]
It looks like the site is abandoned. This critical issue has been broken and untouched for a year.

Looking at it, @scottrice created a repo a couple weeks ago...But he's ignoring anything to do with consolegrid.

**Comment by wyacteen** [comment]
Hey everyone. I got a little bored the other day and played around with getting ConsoleGrid running on my local machine with the hopes of stringing together an alternative solution. I managed to get the site running but of course it doesn't have the data yet. I wrote a script to scrape all of the console and game data into a large JSON file (~5 mb). This wasn't too difficult because thankfully consolegrid doesn't store the images locally (it just provides links to imgur), so I didn't need to actually download the images.

@nateberkopec, do you (or anyone else) have any interest in picking this up or helping out?  I haven't yet figured out how to get the scraped data into the site's database. I've got no experience with ruby so I haven't done that yet.  This is what I think is left to do in order to get this working again, assuming that access to the original consolegrid.com isn't going to happen: 

- Figure out a place to host this.
- Get a new domain name (like perhaps alt-consolegrid.com)
- Getting set up with imgur's API (haven't looked at the code too deeply yet, but I'm assuming you would need some sort of credentials/key in order to create new images)
- Modify the Ice code so that it points to alt-consolegrid.com or modify it so it can get the URL from a config file.

I'm happy to pass along the scraped data to anyone who might be interested in picking this up.  If I get some time, I'll work on a script to get this data into the site's db.

**Comment by nateberkopec** [comment]
@wyacteen happy to help.

**Comment by wyacteen** [comment]
Thanks @nateberkopec!

I just forked this project: https://github.com/wyacteen/ConsoleGrid

Nothing new to the site code, but I did add the scraper program and data that it got (see the consolegrid_scraper directory.

**Comment by wyacteen** [comment]
Update:
I managed to seed the site with *most* of the original data, including all the consoles, games, and images.  I was also able to fix image uploading by using a different gem for imgur access and getting new credentials.  I suspect that the problem with the consolegrid site was that the api calls being made were no longer supported by imgur. See the blog post  about imgur dropping support for old API versions [here](https://blog.imgur.com/2016/01/11/imgur-api-version-3/), which coincides with the timing of the first bug report.

 Here are some screenshots of the site running on my local machine:
[upload successful](http://imgur.com/jz4hPGm)
[new test image](http://imgur.com/gvl8v3r)

The only thing that I know doesn't work currently is voting.  The site crashes when I try to vote. It seems to keep track how I voted after I refresh the page though...  Not sure what the issue is.

**Comment by dkcas11** [comment]
This is awesome @wyacteen! Thanks for your work, would be great if you can fix it :)

**Comment by wyacteen** [comment]
Update:
Just so we don't all go too far off the rails (accidental pun) unnecessarily, if @scottrice is still following this, it looks like all that needs to be done in order to fix the uploading issue is to update the Picture model code to use the latest imgur API.

If that's not going to happen, I did manage to fix the previously noted voting issue.  If anyone is interested in playing around with the latest code that I'm working on it can be found [here](https://github.com/wyacteen/ConsoleGrid/tree/seed-db-with-scraped-data)

Also, I have temporarily set up the working site with the scraped data on my local machine.  I'd appreciate it if people could test it out and let me know if you find any issues.   Here is the address: http://altconsolegrid.mooo.com:3000

Assuming there is nothing major blocking, the next steps are:
- Figure out where and how to deploy the site.
- Update the Ice code to allow it to use the new consolegrid site.

**Comment by casperriboe** [comment]
I will try and do some testing either tonight or tomorrow. I can also contribute to the hosting fees if needed/wanted.

**Comment by nateberkopec** [comment]
@wyacteen Easiest would probably be Heroku. A hobby dyno ($7/mo) plus Mysql ($10) plus a Solr instance ($15?) is all the app would need, I think. My email is on my Github profile if you want to set up a time and we can do it together. I think with all the people using ConsoleGrid we can pull together ~$50/month on a Patreon or something.

**Comment by nateberkopec** [comment]
Also if we switched to Postgres in the long-term, we could eliminate the Solr instance and save on that as well.

**Comment by wyacteen** [comment]
Thanks for the offer to help @nateberkopec!  Just to make my intentions clear, I don't want be in control of the site.  I don't think I'd be a good candidate given my inability to commit time consistently.  I was hoping we could figure something out where perhaps a few enthusiastic users took over. I'm happy to contribute money and time when I can, but I don't want to run the show.

At the risk of sounding cheap, $50/mo sounds kind of expensive to host an app like this.  I'm not expert so I could be way off.  From the looks of it, Heroku seems pretty swank.

**Comment by dkcas11** [comment]
So far everything i've done has worked smoothly.
Created a new user, uploaded a new image, voted for it. Created a new game and added a picture, also voted for it. 
No issues.

**Comment by nateberkopec** [comment]
@wyacteen it's either $50/month in heroku or dozens of hours rolling your own stuff IMO. the time tradeoff would be worth it for me, considering I've never actually deployed mysql or Solr before.

I'll see if I can find some time after Railsconf to set it up.

**Comment by wyacteen** [comment]
@nateberkopec yeah I've done some things with mysql before but never with solr.  We can see how it goes with Heroku.  My concerns about the cost are mostly in consideration of keeping the site running long term.

On a technical note, I'm wondering if anyone has thoughts on the need to 'copy' existing consolegrid images.  The scraper I wrote, simply copies the URLs over, but the existing images are still associated with a different imgur account than the new site would be using.  I'm not sure if there is a risk of the original images becoming inaccessible somehow (like if the original account is closed or made private). If that's possible, we'll need to make copies of the images and upload them to the new account.  That would mean there is a little more work to do before deploying.  Imgur seems to indicate that images can be made private, so if @scottrice ever does that we would lose access to all of the original images.

Oh one more glitch I noticed is that the page navigation control for the games page is missing styling.  That's not critical though.

**Comment by nateberkopec** [comment]
@wyacteen yeah I wouldn't trust imgur either. one option would be to move to Amazon S3 and then throw Cloudflare in front of it. S3 would probably cost a few $ a month but Cloudflare is free.

**Comment by wyacteen** [comment]
@nateberkopec thanks, that might be a better long-term option.   I think for now the best approach would be to get the existing images into the new imgur account so at least we won't lose control of them. I'm not sure when I can get to writing that code, though it shouldn't be too difficult.

**Comment by aaronth07** [comment]
Still not working for me.

**Comment by dkcas11** [comment]
FYI consolegrid.com is now "down" i.e. the domain has expired.

**Comment by wyacteen** [comment]
Hey everyone.  A few updates:

- I gave up trying to get the original rails site up and running because hosting costs were going to be just too high in my opinion.
- Instead, I spent some time creating a new site using php. This site is up and can be found at http://consolegridpics.com
- The project page for the new site can be found at: https://github.com/wyacteen/php_console_grid so feel free to issue bug reports or make code contributions (particularly the UI).
- I forked the Ice project and added support for a new picture provider that points to the new consolegridpics.com site.  You can find that at https://github.com/wyacteen/Ice 

Unfortunately, I have not yet been able to make a windows executable for the new Ice project. So if you want to run it you'll need python (version 2.7 not 3.0+) installed and need to follow the installation instructions on the project's github page. Sorry for this but I've never worked with python before so this is new to me. If anyone wants to help out with that it would be awesome.

**Comment by wyacteen** [comment]
Hello again. A couple of things I didn't realize:

1. You can just update the config.txt file within your _**Ice**_ directory to get it to work with the new site. To do this change the 'Source' entry to point to consolegridpics.com:
   change this **Source=consolegrid.com/api/top_picture**
                to    **Source=consolegridpics.com/api/top_picture**

Note: This assumes that you're using the downloadable executable version of Ice rather than the version from the source which no longer allows for a configurable source.

2.  Someone beat me to the punch!  Months ago, someone made a consolegrid.com replacement, http://retrogaming.cloud  Pretty snazzy site in my opinion. It does have a couple of glitches though with Ice (Ice chokes when it can't find an image from retrogaming)
